pinot-core/src/main/java/org/apache/pinot/core/util/IngestionUtils.java

```diff
@@ -0,0 +1,93 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.core.util;
+
+import com.google.common.collect.Sets;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+import javax.annotation.Nullable;
+import org.apache.pinot.core.data.function.FunctionEvaluator;
+import org.apache.pinot.core.data.function.FunctionEvaluatorFactory;
+import org.apache.pinot.spi.config.table.IngestionConfig;
+import org.apache.pinot.spi.data.FieldSpec;
+import org.apache.pinot.spi.data.Schema;
+import org.apache.pinot.spi.data.readers.GenericRow;
+
+
+/**
+ * Utility methods for extracting source and destination fields from ingestion configs
+ */
+public class IngestionUtils {
+
+  /**
+   * Extracts all fields required by the {@link org.apache.pinot.spi.data.readers.RecordExtractor} from the given TableConfig and Schema
+   * Fields for ingestion come from 2 places:
+   * 1. The schema
+   * 2. The ingestion config in the table config. The ingestion config (e.g. filter) can have fields which are not in the schema.
+   */
+  public static Set<String> getFieldsForRecordExtractor(IngestionConfig ingestionConfig, Schema schema) {
+    Set<String> fieldsForRecordExtractor = new HashSet<>();
+    fieldsForRecordExtractor.addAll(getFieldsFromIngestionConfig(ingestionConfig));
```

Comment:
Recommend passing the set into the helper method to prevent creating multiple sets.

Reply:
done